bundles/tools.vitruv.testutils.changevisualization/src/tools/vitruv/testutils/changevisualization/ChangeVisualizationUI.java

```diff
@@ -0,0 +1,385 @@
+	 */
+	public static final Font DEFAULT_TEXTFIELD_FONT = createFont("TextField.font", 16, Font.PLAIN);
+
+	/**
+	 * Default font used for textAreas
+	 */
+	public static final Font DEFAULT_TEXTAREA_FONT = createFont("TextArea.font", 16, Font.PLAIN);
+
+	/**
+	 * Default font used for checkBoxes
+	 */
+	public static final Font DEFAULT_CHECKBOX_FONT = createFont("CheckBox.font", 16, Font.PLAIN);
+
+	/**
+	 * Default font used for tabbed pane (titles)
+	 */
+	public static final Font DEFAULT_TABBED_PANE_FONT = createFont("TabbedPane.font", 14, Font.BOLD);
+
+	/**
+	 * Default font used for menusd
+	 */
+	public static final Font DEFAULT_MENU_FONT = createFont("Menu.font", 16, Font.PLAIN);
+
+	/**
+	 * Default font used for menuItems
+	 */
+	public static final Font DEFAULT_MENUITEM_FONT = createFont("MenuItem.font", 16, Font.PLAIN);
+
+	private final ChangeVisualizationDataModel changeVisualizationDataModel;
+
+	private JTabbedPane tabbedPane;
+
+	public ChangeVisualizationUI(ChangeVisualizationDataModel changeVisualizationDataModel) {
+		super("Vitruv Change Visualization");
+		this.changeVisualizationDataModel = changeVisualizationDataModel;
+		initializeWindow();
+		initializeContents();
+		initializeFileMenu();
+		setupMonitoringForMonitoredRepositoryChanges(changeVisualizationDataModel);
+	}
+
+	private void initializeContents() {
+		tabbedPane = new JTabbedPane();
+		tabbedPane.setFont(ChangeVisualizationUI.DEFAULT_TABBED_PANE_FONT);// used just for the title
+		setContentPane(tabbedPane);
+	}
+
+	private void initializeWindow() {
+		setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+		int screenWidth = java.awt.Toolkit.getDefaultToolkit().getScreenSize().width;
+		int screenHeight = java.awt.Toolkit.getDefaultToolkit().getScreenSize().height;
+		setSize(Math.min(screenWidth - 30, 1890), Math.min(screenHeight - 60, 1020));
+		setLocationRelativeTo(null);
+	}
+
+	private void setupMonitoringForMonitoredRepositoryChanges(ChangeVisualizationDataModel changeVisualization) {
+		changeVisualization.registerMonitoredRepositoryAddedListener(this);
+		tabbedPane.addContainerListener(new ContainerListener() {
+			@Override
+			public void componentAdded(ContainerEvent e) {
+			}
+
+			@Override
+			public void componentRemoved(ContainerEvent e) {
+				if (e.getChild() instanceof ChangesTab) {
+					ChangesTab tab = (ChangesTab) e.getChild();
+					changeVisualization.removeMonitoredRepository(tab.getTitle());
+				}
+			}
+		});
+		tabbedPane.addChangeListener(new ChangeListener() {
+			@Override
+			public void stateChanged(ChangeEvent e) {
+				for (Component component : tabbedPane.getComponents()) {
+					if (component instanceof ChangesTab) {
+						((ChangesTab) component).setEnabled(tabbedPane.getSelectedComponent() == component);
+					}
+				}
+			}
+		});
+	}
+
+	@Override
+	public void setVisible(boolean isVisible) {
+		super.setVisible(isVisible);
+		if (isVisible) {
+			setAlwaysOnTop(true);
+			setAlwaysOnTop(false);
```

Comment:
Either this is a bug or a weird implementation of `toFront()`.

Reply:
As far as I understand, the difference is that `toFront()` is restricted to the VM while `setAlwaysOnTop` is for the complete platform (i.e., the OS).

Reply:
@tsaglam Should we open an issue for documenting this?

Reply:
Ah, okay, with this explanation in mind I think it is fine without an issue.